server/chat-plugins/datasearch.js

```diff
@@ -1212,12 +1255,22 @@ function runMovesearch(target, cmd, canAll, message) {
 	}
 	if (results.length > 1) {
 		results.sort();
+		if (sort) {
+			let prop = sort.slice(0, -1);
+			let direction = sort.slice(-1);
+			results.sort((a, b) => {
+				let move1prop = dex[toId(direction === '+' ? a : b)][prop], move2prop = dex[toId(direction === '+' ? b : a)][prop];
+				return ~~move1prop - ~~move2prop; // Incase they are boolean valuess, they are converted to 0 or 1
```

Comment:
I would probably do the boolean-to-int conversion explicitly, instead of the implicit cast to int here. You don't want to cast to int if it's already a Number anyway; that will screw up fractions like sorting by height/weight.